server/remote/common.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package remote
+
+import "time"
+
+type DateTime struct {
+	DateTime string `json:"dateTime"`
+	TimeZone string `json:"timeZone,omitempty"`
+}
+
+func NewDateTime(t time.Time) *DateTime {
+	return &DateTime{
+		DateTime: t.Format("2006-01-02T15:04:05.999999999"), // time.RFC3339Nano sans timezone
```

Comment:
Do you think we should store these date formats somewhere? I saw some hardcoded formats elsewhere as well.